### PR TITLE
RTEMIS-164: Modified the title, success message and overview form.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-164

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Modified the title, success message and overview form for term.

### CODE REVIEW

File: modules/rte_mis_core/rte_mis_core.module [hunk 1]
```diff
@@ -115,6 +117,36 @@ function rte_mis_core_form_alter(&$form, FormStateInterface $form_state, string
     ];
 
   }
+  elseif ($form_id === 'taxonomy_overview_terms') {
+    // Get the vocabulary ID from the form state.
+    $vid = $form_state->getBuildInfo()['args'][0];
```
Comment (kunal-sin) [inline modules/rte_mis_core/rte_mis_core.module:122]: Can we add a safe check here?

File: modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      case 'location':
+        $title = 'Add location';
+        break;
+
+      case 'school':
+        $title = 'Add School';
+        break;
+
+      case 'location_schema':
+        $title = 'Add location schema';
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php:30]: `Add Location Schema`

File: modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  /**
+   * {@inheritDoc}
+   */
+  public function addForm(VocabularyInterface $taxonomy_vocabulary) {
+    $term = $this->entityTypeManager()->getStorage('taxonomy_term')->create(['vid' => $taxonomy_vocabulary->id()]);
+    $form = $this->entityFormBuilder()->getForm($term);
+
+    switch ($taxonomy_vocabulary->id()) {
+      case 'location':
+        $title = 'Add location';
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php:22]: `Add Location`

File: modules/rte_mis_state/src/Routing/RouteSubscriber.php [hunk 4]
```diff
@@ -38,6 +38,22 @@ protected function alterRoutes(RouteCollection $collection) {
...
+      $termAddDefaults['_controller'] = 'Drupal\rte_mis_core\Controller\OverrideTaxonomyController::addForm';
+      $termAddPage->setDefaults($termAddDefaults);
+    }
+
+    $termEditPage = $collection->get('entity.taxonomy_term.edit_form');
+    if ($termEditPage instanceof Route) {
+      // Update the title for the route.
+      $termEditDefaults = $termEditPage->getDefaults();
+      $termEditDefaults['_title'] = 'Edit';
+      $termEditPage->setDefaults($termEditDefaults);
```
Comment (kunal-sin) [inline modules/rte_mis_state/src/Routing/RouteSubscriber.php:55]: @RohanSingh-inno We discussed a different approach yesterday.

File: modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php (hunk 2 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php:30]: ok

File: modules/rte_mis_core/rte_mis_core.module (hunk 1 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_core/rte_mis_core.module:122]: ok

File: modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php (hunk 3 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php:22]: ok

File: modules/rte_mis_state/src/Routing/RouteSubscriber.php (hunk 4 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_state/src/Routing/RouteSubscriber.php:55]: That won't work as there are two separate events in event subscriber and route subscriber, we have **RequestEvent $event** in event subscriber and **RouteBuildEvent $event**. The later ultimately returns the collections. Referrence: [RouteBuildEvent](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Routing%21RouteBuildEvent.php/class/RouteBuildEvent/10)

File: modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        break;
+
+      case 'location_schema':
+        $title = 'Add Location Schema';
+        break;
+
+      default:
+        $title = 'Add Term';
+        break;
+    }
```
Comment (jaswinsingh) [inline modules/rte_mis_core/src/Controller/OverrideTaxonomyController.php:36]: @RohanSingh-inno Please wrap the string in `t()`

File: modules/rte_mis_state/src/Routing/RouteSubscriber.php [hunk 6]
```diff
@@ -38,6 +38,22 @@ protected function alterRoutes(RouteCollection $collection) {
...
+      $termAddDefaults = $termAddPage->getDefaults();
+      $termAddDefaults['_controller'] = 'Drupal\rte_mis_core\Controller\OverrideTaxonomyController::addForm';
+      $termAddPage->setDefaults($termAddDefaults);
+    }
+
+    $termEditPage = $collection->get('entity.taxonomy_term.edit_form');
+    if ($termEditPage instanceof Route) {
+      // Update the title for the route.
+      $termEditDefaults = $termEditPage->getDefaults();
+      $termEditDefaults['_title'] = 'Edit';
```
Comment (jaswinsingh) [inline modules/rte_mis_state/src/Routing/RouteSubscriber.php:54]: @RohanSingh-inno Same here.

**Pull request merged**